### PR TITLE
Feature/database

### PR DESCRIPTION
**Pull request by Jnf14**
Required coverage percentage not achieved yet, but big feature to merge.

### CODE REVIEW

**Review by Arthelh — CHANGES_REQUESTED** [review]
Enlevez le fichier google-services.json du git. Et, ligne 12 de ce fichier, est-ce qu'il faudrait pas mettre "package_name": "com.ncnf" plutôt?

**Comment by Jnf14** [comment]
> Enlevez le fichier google-services.json du git.

Pas besoin pour cirrus ? Il y a tout ce qui est Firebase là dedans !

 >Et, ligne 12 de ce fichier, est-ce qu'il faudrait pas mettre "package_name": "com.ncnf" plutôt?

Si tu regardes plus bas on a une autre app qui tourne sur Firebase avec l'ancien nom. On ne l'a pas encore supprimé "Au cas où".

**Review by Arthelh — APPROVED** [review]

File: app/src/main/java/com/ncnf/user/PrivateUser.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import static com.ncnf.Utils.EMPTY_STRING;
+import static com.ncnf.Utils.FIRST_NAME_KEY;
+import static com.ncnf.Utils.FRIENDS_KEY;
+import static com.ncnf.Utils.LAST_NAME_KEY;
+import static com.ncnf.Utils.OWNED_EVENTS_KEY;
+import static com.ncnf.Utils.SAVED_EVENTS_KEY;
+
+public class PrivateUser {
+
+    private DatabaseServiceInterface db;
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/user/PrivateUser.java:32]: Make attributes final

File: app/src/main/java/com/ncnf/user/UserProfileActivity.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+@AndroidEntryPoint
+public class UserProfileActivity extends AppCompatActivity {
+
+    @Inject
+    public PrivateUser user;
+
+    private boolean firstNameChanged = false;
+    private boolean lastNameChanged = false;
+    private boolean birthDateChanged = false;
+    EditText email;
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/user/UserProfileActivity.java:40]: make attributes private and final (except for booleans)

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -6,44 +6,65 @@
     android:layout_height="match_parent"
     tools:context="com.ncnf.main.MainActivity">
 
+    <Button
+        android:id="@+id/mainProfileButton"
+        android:layout_width="200dp"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="112dp"
+        android:onClick="gotToProfile"
+        android:text="Profile"
```
Comment (HyPn0TiiK) [inline app/src/main/res/layout/activity_main.xml:15]: Make text refer to a string in res/value with @string

File: app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -6,44 +6,65 @@
...
+        app:rippleColor="#FFFFFF" />
+
+    <Switch
+        android:id="@+id/connected"
+        android:layout_width="145dp"
+        android:layout_height="30dp"
+        android:layout_marginTop="16dp"
+        android:layout_marginEnd="16dp"
+        android:clickable="false"
+        android:text="Disconnected"
```
Comment (HyPn0TiiK) [inline app/src/main/res/layout/activity_main.xml:29]: Make text refer to a string in res/value with @string

File: app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -6,44 +6,65 @@
...
 
     <Button
-        android:id="@+id/feed_view_button"
+        android:id="@+id/feedViewButton"
         android:layout_width="150dp"
         android:layout_height="40dp"
-        android:layout_marginLeft="90dp"
-        android:layout_marginBottom="204dp"
+        android:layout_marginTop="12dp"
         android:text="FeedView"
```
Comment (HyPn0TiiK) [inline app/src/main/res/layout/activity_main.xml:51]: Make text refer to a string in res/value with @string

File: app/src/main/res/layout/activity_user_profile.xml [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <Button
+        android:id="@+id/logOutButton"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="62dp"
+        android:onClick="logOut"
+        android:text="Log Out"
```
Comment (HyPn0TiiK) [inline app/src/main/res/layout/activity_user_profile.xml:30]: Make text refer to a string in res/value with @string

File: app/src/main/res/layout/activity_user_profile.xml [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="49dp"
+        android:layout_marginEnd="101dp"
+        android:clickable="false"
+        android:ems="10"
+        android:enabled="false"
+        android:gravity="start|top"
+        android:inputType="textMultiLine"
+        android:text="Welcome to your Profile"
```
Comment (HyPn0TiiK) [inline app/src/main/res/layout/activity_user_profile.xml:20]: Make text refer to a string in res/value with @string

File: app/src/main/res/layout/activity_user_profile.xml [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    <EditText
+        android:id="@+id/userProfileEmail"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="42dp"
+        android:layout_marginEnd="168dp"
+        android:clickable="false"
+        android:ems="10"
+        android:enabled="true"
+        android:hint="Email Address"
```
Comment (HyPn0TiiK) [inline app/src/main/res/layout/activity_user_profile.xml:45]: Make text refer to a string in res/value with @string

File: app/src/main/res/layout/activity_user_profile.xml [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+        app:layout_constraintTop_toBottomOf="@+id/welcomeToProfileText" />
+
+    <EditText
+        android:id="@+id/userProfileFirstName"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="32dp"
+        android:layout_marginEnd="168dp"
+        android:ems="10"
+        android:hint="Firstname..."
```
Comment (HyPn0TiiK) [inline app/src/main/res/layout/activity_user_profile.xml:57]: Make text refer to a string in res/value with @string

File: app/src/main/res/layout/activity_user_profile.xml [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+        app:layout_constraintTop_toBottomOf="@+id/userProfileFirstName" />
+
+    <EditText
+        android:id="@+id/userProfileDateOfBirth"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="32dp"
+        android:layout_marginEnd="166dp"
+        android:ems="10"
+        android:hint="Date of birth..."
```
Comment (HyPn0TiiK) [inline app/src/main/res/layout/activity_user_profile.xml:81]: Make text refer to a string in res/value with @string

File: app/src/main/res/layout/activity_user_profile.xml [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+        app:layout_constraintTop_toBottomOf="@+id/userProfileEmail" />
+
+    <EditText
+        android:id="@+id/userProfileLastName"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="32dp"
+        android:layout_marginEnd="167dp"
+        android:ems="10"
+        android:hint="Lastname..."
```
Comment (HyPn0TiiK) [inline app/src/main/res/layout/activity_user_profile.xml:69]: Make text refer to a string in res/value with @string

File: app/src/main/res/layout/activity_user_profile.xml [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+
+    <Button
+        android:id="@+id/userProfileSaveButton"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="332dp"
+        android:layout_marginEnd="38dp"
+        android:enabled="false"
+        android:onClick="saveNewFields"
+        android:text="Save"
```
Comment (HyPn0TiiK) [inline app/src/main/res/layout/activity_user_profile.xml:94]: Make text refer to a string in res/value with @string

File: app/src/main/res/layout/fragment_sign_in.xml [hunk 13]
```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (HyPn0TiiK) [inline app/src/main/res/layout/fragment_sign_in.xml:1]: Make text refer to a string in res/value with @string

File: app/src/main/res/layout/fragment_sign_up.xml [hunk 14]
```diff
@@ -0,0 +1,95 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (HyPn0TiiK) [inline app/src/main/res/layout/fragment_sign_up.xml:1]: Make text refer to a string in res/value with @string

**Review by HyPn0TiiK — APPROVED** [review]

**Review by Arthelh — APPROVED** [review]

**Pull request merged**